src/main/java/com/snowflake/kafka/connector/internal/BufferThreshold.java

```diff
@@ -0,0 +1,213 @@
+   *     stage for snowpipe based implementation or previous buffered records were sent in
+   *     insertRows API of Streaming Snowpipe.
+   * @return true if time based threshold has reached.
+   */
+  public boolean isFlushTimeBased(final long previousFlushTimeStampMs) {
+    final long currentTimeMs = System.currentTimeMillis();
+    return (currentTimeMs - previousFlushTimeStampMs)
+        >= (this.flushTimeThresholdSeconds * SECOND_TO_MILLIS);
+  }
+
+  /** Get flush time threshold in seconds */
+  public long getFlushTimeThresholdSeconds() {
+    return flushTimeThresholdSeconds;
+  }
+
+  /**
+   * Check if provided snowflake kafka connector buffer properties are within permissible values
+   *
+   * @param providedSFConnectorConfig provided by customer
+   * @return true if valid.
+   */
+  public static boolean validateBufferThreshold(
+      Map<String, String> providedSFConnectorConfig, IngestionMethodConfig ingestionMethodConfig) {
+    if (!providedSFConnectorConfig.containsKey(
+        SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC)) {
+      LOGGER.error(
+          Logging.logMessage("{} is empty", SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC));
+      return false;
+    } else {
+      try {
+        long time =
+            Long.parseLong(
+                providedSFConnectorConfig.get(SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC));
+        if (ingestionMethodConfig.equals(IngestionMethodConfig.SNOWPIPE)) {
+          if (time < BUFFER_FLUSH_TIME_SEC_MIN) {
+            LOGGER.error(
+                (Logging.logMessage(
+                    "{} is {}, it should be greater than {}",
+                    SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC,
+                    time,
+                    BUFFER_FLUSH_TIME_SEC_MIN)));
+            return false;
+          }
+        } else {
+          if (time < STREAMING_BUFFER_FLUSH_TIME_MINIMUM_SEC) {
+            LOGGER.error(
+                (Logging.logMessage(
+                    "{} is {}, it should be greater than {}",
+                    SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC,
+                    time,
+                    STREAMING_BUFFER_FLUSH_TIME_MINIMUM_SEC)));
+            return false;
+          }
+        }
+      } catch (Exception e) {
+        LOGGER.error(
+            Logging.logMessage(
+                "{} should be an integer", SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC));
+        return false;
+      }
+    }
+
+    // verify buffer.count.records
+    if (!providedSFConnectorConfig.containsKey(SnowflakeSinkConnectorConfig.BUFFER_COUNT_RECORDS)) {
+      LOGGER.error(
+          Logging.logMessage("{} is empty", SnowflakeSinkConnectorConfig.BUFFER_COUNT_RECORDS));
+      return false;
+    } else {
+      try {
+        long num =
+            Long.parseLong(
+                providedSFConnectorConfig.get(SnowflakeSinkConnectorConfig.BUFFER_COUNT_RECORDS));
+        if (num < 0) {
```

Comment:
Is `0` a valid value?

Reply:
Yeah not a valid value. It should be <= 0 technically, 

even in SinkConnectorConfig this is 
```.define(
            BUFFER_COUNT_RECORDS,
            Type.LONG,
            BUFFER_COUNT_RECORDS_DEFAULT,
            ConfigDef.Range.atLeast(1),
            Importance.LOW,
            "Number of records buffered in memory per partition before triggering Snowflake"
                + " ingestion",
            CONNECTOR_CONFIG,
            1,
            ConfigDef.Width.NONE,
            BUFFER_COUNT_RECORDS)


Sorry, overall I was hesitant to change what was already there but this qualifies for the fix. thanks for catching.

Reply:
Changed this to <=0